### PR TITLE
Poll real health checks

### PR DESCRIPTION
**Pull request by 2uasimojo**
With this PR, we poll [real osde2e health checks](https://github.com/openshift/osde2e/blob/041355675304a7aa371b7fbeea313001036feb75/pkg/common/cluster/clusterutil.go#L211) at one-minute intervals, forever. If they fail, we make sure there's an active silence. If they succeed, we make sure the silence is removed and we exit.

### CODE REVIEW

**Comment by 2uasimojo** [comment]
/hold

Prereq https://github.com/openshift/osde2e/pull/681

File: go.mod
Comment (2uasimojo) [inline go.mod]: This will need to be updated with a new hash once https://github.com/openshift/osde2e/pull/681 lands.

File: main.go
Comment (iamkirkbater) [inline main.go:49]: Is there a reason why we're getting rid of this cluster timestamp check?  I'm not sure off the top of my head how MCC pushes out changes to this job, so we'd want to confirm that it's not going to re-run the job on existing clusters when we push an update.

If it doesn't re-run the job on update, then we're safe to remove this, however if it does re-run the job this could silence an already-ready production cluster if it's experiencing a blip of something updating.

File: main.go
Comment (iamkirkbater) [inline main.go:65]: In osde2e they've found that the cluster can become ready but somewhere like 5-10 minutes later it just starts upgrading, so they've created a loop where they check to see that it's healthy 20 times in a row before continuing.  Should we implement something similar here?  CC: @jeefy for additional context

File: main.go
Comment (2uasimojo) [reply main.go:49]: Yeah, I struggled with that. You're probably right that we should keep this check in place -- but from what @jharrington22 was telling me yesterday, one hour may be way too short. James, what do you think?

Comment (iamkirkbater) [reply main.go:49]: The job _should_ be deployed within one hour via MCC, but really I'd be comfortable putting this to even the first 4 hours or so that a cluster is deployed, that would make testing easier and a bit more flexible.

File: main.go
Comment (jeefy) [reply main.go:65]: :100: -- We've often seen that a cluster will be "healthy" for a period of time then start updating operators. 

In our OSDe2e job runs we require 20 "Clean" healthchecks *_in a row_* before moving forward. 

That number was arbitrary, but I've even recently seen a cluster having 12 clean runs then going back to 0.

Comment (2uasimojo) [reply main.go:65]: > 20 "Clean" healthchecks __in a row__

over what time period?

File: main.go
Comment (2uasimojo) [reply main.go:49]: Roger wilco, I'll do that. Mebbe I'll add support for an env var so we can configure it (including for testing).

File: main.go
Comment (2uasimojo) [reply main.go:65]: ...looks like [30s](https://github.com/openshift/osde2e/blob/7e201f679bbe1a78b67701df77ebb69bcf768957/pkg/common/cluster/clusterutil.go#L102) x [20 by default](https://github.com/openshift/osde2e/blob/7e201f679bbe1a78b67701df77ebb69bcf768957/pkg/common/config/config.go#L540).

Which means that we're going to continue to silence alerts for a full 10 minutes after the cluster is healthy.

Comment (iamkirkbater) [reply main.go:65]: I think that's fair.  At this point in the cluster lifecycle the cluster shouldn't really have much if any customer workload on it that this should disturb.

Comment (iamkirkbater) [reply main.go:65]: @jeefy yesterday when I was working on the CI Watcher stuff I was looking at a cluster that had never became "healthy" after an upgrade and it was getting to 15 and even 18 before something would roll again and reset the clock.

Comment (jeefy) [reply main.go:65]: Yep.... :fire:

File: go.mod
Comment (2uasimojo) [reply go.mod]: Done.

File: main.go
Comment (2uasimojo) [reply main.go:49]: This is done via 060f06b.

File: main.go
Comment (2uasimojo) [reply main.go:65]: Still working on this.

**Comment by 2uasimojo** [comment]
> /hold
> 
> Prereq [openshift/osde2e#681](https://github.com/openshift/osde2e/pull/681)

Prereq merged.

But still /hold (not that it's doing anything) for [multiple successful health checks](https://github.com/iamkirkbater/osd-cluster-ready-job/pull/1#discussion_r568616600)

File: main.go
Comment (2uasimojo) [reply main.go:65]: Done via 27771f4.

**Comment by 2uasimojo** [comment]
> But still /hold (not that it's doing anything) for [multiple successful health checks](https://github.com/iamkirkbater/osd-cluster-ready-job/pull/1#discussion_r568616600)

This is done.

/hold cancel

(FWIW)

**Review by iamkirkbater — APPROVED** [review]
LGTM.

I know @jharrington22 said he wanted to review, so once he says this is good go ahead and merge it and build and push as you would.

File: README.md
Comment (jharrington22) [inline README.md:13]: Non blocker but its not only the day 2 configuration keeping these clusters from being ready. ClusterOperators is still observing operators from being ready as they transition through different phases while configuration from each is loaded.

File: README.md
Comment (jharrington22) [inline README.md:21]: :+1:

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by 2uasimojo** [comment]
@iamkirkbater I don’t think I can merge this. Would you please do the honors?

**Pull request merged**
